Spec review fixes (#1112)
riscv/riscv-cheri: src/cheri/riscv-hybrid-integration.adoc

NOTE: While the <<p_bit>> can be read or written explicitly using <<GCMODE>> and <<SCMODE>>, these instructions do not change the current execution mode.
The mode changes when a capability is installed into <<pcc>> (e.g., via an indirect jump, see <<sec_changing_cheri_execution_mode>>).

The <<p_bit>> is only valid in capabilities that grant <<x_perm>> (see <<zyhybrid-clrperm-rules>>).

> Capabilities that do not grant <<x_perm>> therefore do not carry a meaningful <<p_bit>>, and <<SCMODE>> does not modify their <<p_bit>>.

Reviewer: ```suggestion
Capabilities that do not grant <<x_perm>> therefore do not carry a meaningful <<p_bit>>, and so <<SCMODE>> cannot modify it.
```

Author: I think this makes it more ambiguous, I'd avoid `it`

Author: ```suggestion
Capabilities that do not grant <<x_perm>> therefore do not carry a meaningful <<p_bit>>, so <<SCMODE>> cannot not modify their <<p_bit>>.
```
